addons/sale/models/sale.py

```diff
@@ -577,6 +577,18 @@ def _compute_qty_delivered_updateable(self):
         for line in self:
             line.qty_delivered_updateable = (line.order_id.state == 'sale') and (line.product_id.track_service == 'manual') and (line.product_id.expense_policy == 'no')
 
+    @api.depends('invoice_lines.invoice_id.state', 'invoice_lines.invoice_id.refund_invoice_id.state')
```

Comment:
_compute_invoice_amount method is not clean.

can you test with this example ?

Sale order
================
product1 -- 300Rs
product2 -- 400Rs
product3 -- 200Rs


Advance Invoice1
==========
product1 -- 100Rs


Invoice2
===========
product1 -- 200Rs


Invoice3
==========
product2 -- 400Rs
product3 -- 100 Rs



Invoice4
=========
product3  -- 100 Rs

Invoice5 (refund of invoice3)
=========
product2  -- 100Rs
product3  -- 50 Rs

so what's invoiced amount of product2 and product3 in sale order?